### PR TITLE
Avoid reattaching observers to destroyed IceStorm topics

### LINKED ISSUE #5996: IceStorm TopicImpl::updateObserver should check not destroyed
This `!_destroyed` guard is the right call — note that the persistent `TopicImpl::updateObserver` doesn't have it, so a metrics-view update between destroy and reap re-attaches an observer to a destroyed persistent topic. Not this PR's problem, but worth mirroring the guard back into `TopicImpl::updateObserver` as a small follow-up.

_Originally posted by @bernardnormier in https://github.com/zeroc-ice/ice/pull/5966#discussion_r3580851218_

### PR DESCRIPTION
**Pull request by pepone**
## Summary

- prevent persistent IceStorm from reattaching a metrics observer to a destroyed topic while it waits to be reaped
- document the metrics lifecycle fix in the IceStorm changelog

## Testing

- make -C cpp -j8 IceStormService
- python3 allTests.py IceStorm/createOrRetrieve

Fixes #5996

### CODE REVIEW

**Review by bernardnormier — APPROVED** [review]
The fix is correct: `_destroyed` is only written under `_subscribersMutex` and `updateObserver` reads it under the same lock, and this mirrors the `!_destroyed` guard the transient implementation already has.

Two comments:

- I'd drop the changelog entry. The trigger is a runtime metrics-view update landing precisely in the destroy→reap window, and the effect is a transiently stale metrics entry — hardening-grade, not something a real application would notice.

- The replica path has a related asymmetry: `TopicImpl::destroy` (master) explicitly detaches `_observer`, but `observerDestroyTopic` never does — `destroyInternal` doesn't touch `_observer`. So on a replica, a destroyed topic's observer stays attached to the current metrics view until the topic is reaped. This guard correctly prevents resurrection on a view update, but a `_observer.detach()` in `observerDestroyTopic` (or a follow-up) would make the destroy behavior consistent on replicas.

File: cpp/src/IceStorm/TopicI.cpp [hunk 1]
```diff
@@ -1174,7 +1174,8 @@ TopicImpl::updateObserver()
 {
     lock_guard lock(_subscribersMutex);
 
-    if (_instance->observer())
+    // Don't reattach the observer of a topic that was destroyed but not reaped yet.
+    if (!_destroyed && _instance->observer())
```
[the remark this answers is not in the document]
Comment (pepone) [reply cpp/src/IceStorm/TopicI.cpp:1180]: 🤖 **Claude:** Skipping a dedicated metrics regression test here. This guard mirrors the one `TransientTopicImpl::updateObserver` already carries, and a faithful test would have to enable an IceMX metrics view and race the reap timer to catch a transiently-stale entry — disproportionate for a one-line hardening fix that duplicates an existing guard.

**Comment by pepone** [comment]
🤖 **Claude:** Thanks Bernard.

- Dropped the changelog fragment — agreed, this is hardening-grade.
- Added `_observer.detach()` in `observerDestroyTopic` so a destroyed-but-unreaped topic leaves the metrics view immediately on replicas too, matching the master `destroy` path instead of lingering until reaping.

Done in f13fe19066.
